BUG: `docker compose` up fails on fresh clone with EACCES during `@agor/core` tsup build (cannot write `tsup.config.bundled_*.mjs` under `/app`)

Running the documented Docker/Compose dev setup on a fresh clone fails during the `@agor/core` build step. The dev container can’t write a temporary tsup config next to `packages/core/tsup.config.*`, causing:

```bash
Error: EACCES: permission denied, open '/app/packages/core/tsup.config.bundled_<hash>.mjs'
ERR_PNPM_RECURSIVE_RUN_FIRST_FAIL  @agor/core@0.1.0 build: `tsup`
Exit status 1
```

A few lines earlier the startup script prints:

```bash
husky - install command is DEPRECATED
fatal: not in a git directory
```

(the Husky messages appear to be side-noise; the fatal error is the EACCES above).

**Why this matters**:
The Development Guide recommends Docker/Compose for the quickest setup; having the default path fail blocks new contributors. ([agor][1])

**Steps to reproduce**:

1. Fresh clone:

   ```bash
   git clone https://github.com/preset-io/agor && cd agor
   ```

2. Start dev stack:

   ```bash
   docker compose down -v # just to ensure a clean state
   docker compose up --build
   ```

3. Observe the `EACCES` error when `tsup` builds `@agor/core`.

**Expected**:
Dev stack builds and the UI comes up at `http://localhost:5173`.

**Actual**:
Build fails with `EACCES` while trying to write `tsup.config.bundled_*.mjs` under `/app/packages/core`.

**Environment**:

```bash
# fill these in
git rev-parse --short HEAD
# a64c7db                                                                                                      → 0.01s
docker compose version
# Docker Compose version 2.40.3                                                                                → 0.07s
docker --version
# Docker version 28.5.2, build ecc694264d
docker info --format 'Server: {{.ServerVersion}}  OS: {{.OperatingSystem}}'
# Server: 28.5.2  OS: Arch Linux
uname -a
# Linux thearchdimension 6.12.57-1-lts #1 SMP PREEMPT_DYNAMIC Sun, 02 Nov 2025 15:08:33 +0000 x86_64 GNU/Linux
# if on WSL2, also echo your path (e.g., /home/... vs /mnt/c/...)
pwd
# /home/rashino/repos/agor
# host dir ownership and perms:
ls -ld .
# drwxr-xr-x - rashino 15 Nov 14:04  ./
ls -ld packages/core
# drwxr-xr-x - rashino 15 Nov 12:55  packages/core/                                                           → 0.02s
```

**Workarounds tested**:

- ✅ Works if I run the container as my host UID/GID and bind‑mount the repo (override):

  ```yaml
  # docker-compose.override.yml
  services:
    agor-dev:
      user: "${UID:-1000}:${GID:-1000}"
      volumes:
        - .:/app:rw
        - agor_node_modules:/app/node_modules
  volumes:
    agor_node_modules:
  ```

  And then:

  ```bash
  # Linux/macOS
  printf "UID=%s\nGID=%s\n" "$(id -u)" "$(id -g)" > .env
  docker compose up --build
  ```

  After this, the build succeeds and the app runs.
- ✅ Also works if I run as root in the container (not ideal):

  ```yaml
  services:
    agor-dev:
      user: "0:0"
  ```

- ⛔ Fails reproducibly with the default `docker compose up` on a clean clone.

**Hypothesis**:
The dev image runs as a non‑root user but `/app` (the code copied into the image at build time) is owned by root with `755`, so `tsup` can’t emit the bundled config file next to `tsup.config.*`. Binding the repo with a writable user mapping, or `chown`‑ing `/app` in the image, resolves it.

[1]: https://agor.live/guide/development "Development Guide - agor"

--- a/.env.example
+++ b/.env.example
@@ -1,0 +1,56 @@
+# Agor Docker Development Environment Configuration
+# Copy this file to .env and customize for your setup
+
+# ============================================================================
+# Port Configuration
+# ============================================================================
+
+# Daemon API port (default: 3030)
+DAEMON_PORT=3030
+
+# UI development server port (default: 5173)
+UI_PORT=5173
+
+# CORS origin (default: * for dev, set specific origins in prod)
+# CORS_ORIGIN=http://localhost:5173
+
+# ============================================================================
+# Development Data Seeding
+# ============================================================================
+
+# Seed development fixtures on first startup (default: false)
+# Creates: Agor repo (https://github.com/preset-io/agor.git) + test-worktree
+SEED=false
+
+# ============================================================================
+# AI Agent API Keys (Optional)
+# ============================================================================
+
+# Anthropic Claude API key for Claude Code sessions
+# ANTHROPIC_API_KEY=sk-ant-...
+
+# OpenAI API key for Codex sessions
+# OPENAI_API_KEY=sk-...
+
+# Google Gemini API key for Gemini sessions
+# GEMINI_API_KEY=...
+
+# ============================================================================
+# Advanced: Host UID/GID Mapping (Linux Only)
+# ============================================================================
+
+# On Linux, you can optionally run the container as your host user to avoid
+# the automatic permission fix and ensure container-created files match your
+# host user ownership. Uncomment these lines and set to your UID/GID:
+#
+# Get your UID/GID with: id -u && id -g
+# UID=1000
+# GID=1000
+#
+# Then create docker-compose.override.yml with:
+#   services:
+#     agor-dev:
+#       user: "${UID:-1000}:${GID:-1000}"
+#
+# Note: On macOS/Windows, Docker Desktop handles UID mapping automatically.
+# This setting is unnecessary (and may cause issues) on those platforms.

--- a/docker-compose.yml
+++ b/docker-compose.yml
@@ -31,7 +31,7 @@ services:
       - OPENAI_API_KEY=${OPENAI_API_KEY:-}
       - GEMINI_API_KEY=${GEMINI_API_KEY:-}
     volumes:
-      # Mount entire repo for hot-reload (but exclude node_modules)
+      # Mount entire repo for hot-reload (but exclude node_modules and build outputs)
       - .:/app
       # Exclude all node_modules directories (use container's versions)
       - /app/node_modules
@@ -39,6 +39,9 @@ services:
       - /app/apps/agor-cli/node_modules
       - /app/apps/agor-ui/node_modules
       - /app/packages/core/node_modules
+      # Exclude @agor/core build output (built fresh in container during entrypoint)
+      # This prevents empty/stale host dist from overwriting container's fresh build
+      - /app/packages/core/dist
       # Persist entire home directory (database, config, agent auth tokens, etc.)
       - agor-home:/home/agor
       # Mount SSH keys for git authentication (dev only)

--- a/docker/README.md
+++ b/docker/README.md
@@ -182,7 +182,33 @@ volumes:
   - ~/.ssh:/home/agor/.ssh:ro
 ```
 
-This is already configured in `docker-compose.yml` (commented out by default).
+This is already configured in `docker-compose.yml`.
+
+### Host UID/GID Mapping (Linux)
+
+On Linux, you can run the container as your host user to avoid permission issues entirely:
+
+```bash
+# Create .env file with your UID/GID
+printf "UID=%s\nGID=%s\n" "$(id -u)" "$(id -g)" > .env
+
+# Create docker-compose.override.yml
+cat > docker-compose.override.yml <<'EOF'
+services:
+  agor-dev:
+    user: "${UID:-1000}:${GID:-1000}"
+EOF
+
+# Start normally
+docker compose up
+```
+
+**When to use this:**
+
+- You're on Linux and want to avoid the automatic `chown` on startup
+- You want container-created files to match your host user ownership
+
+**Note:** On macOS/Windows, Docker Desktop handles UID mapping automatically - this override is unnecessary.
 
 ### Custom agor-live Version (Prod)
 
@@ -210,12 +236,19 @@ DAEMON_PORT=3031 docker compose up
 
 ### Volume Permission Issues
 
-If you see permission errors:
+The entrypoint script automatically fixes permissions on bind-mounted directories during startup. This resolves issues where host-mounted files have different ownership than the container user.
+
+**If you still see `EACCES: permission denied` errors:**
 
 ```bash
-# Fix permissions (runs automatically in entrypoint)
-docker compose exec agor-dev sudo chown -R agor:agor /home/agor/.agor
+# Manually fix permissions (already runs automatically in entrypoint)
+docker compose exec agor-dev sudo chown -R agor:agor /app
+
+# Or restart container (permissions are fixed on startup)
+docker compose restart agor-dev
 ```
+
+**Note for Linux users:** On Linux, bind mounts preserve exact host UID/GID. If you prefer to avoid the automatic permission fix and match your host user, use the UID/GID override (see Advanced Usage below).
 
 ### Rebuild from Scratch
 

--- a/docker/docker-entrypoint.sh
+++ b/docker/docker-entrypoint.sh
@@ -3,6 +3,11 @@ set -e
 
 echo "🚀 Starting Agor development environment..."
 
+# Fix permissions on bind-mounted source directory
+# (host-mounted files may have different ownership than container user)
+echo "🔧 Fixing source directory permissions..."
+sudo chown -R agor:agor /app
+
 # Dependencies are installed during Docker build and node_modules are excluded from volume mount
 # Just verify they exist, don't reinstall unless something is actually missing
 if [ ! -d "/app/node_modules" ]; then

